syntax = "proto3";

package littlehorse;

import "google/protobuf/timestamp.proto";
import "object_id.proto";

option go_package = ".;model";
option java_multiple_files = true;
option java_package = "io.littlehorse.sdk.common.proto";
option csharp_namespace = "LittleHorse.Common.Proto";

message VariableValue {
  reserved 1;
  oneof value {
    string json_obj = 2;
    string json_arr = 3;
    double double = 4;
    bool bool = 5;
    string str = 6;
    int64 int = 7;
    bytes bytes = 8;
  }
}

message Variable {
  VariableId id = 1;
  VariableValue value = 2;
  google.protobuf.Timestamp created_at = 3;
  WfSpecId wf_spec_id = 4;
}
